CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+3.  When the feature or fix is completed you should open a
+    [Pull Request](https://help.github.com/articles/using-pull-requests) on GitHub.
+4.  The Pull Request should be reviewed by other maintainers (as many as feasible/practical),
+    among which at least one core developer.
+    Independent contributors can also participate in the review process,
+    and are encouraged to do so.
+5.  After the review, you should resolve issues brought up by the reviewers as needed
+    (amending or adding commits to address reviewers' comments), iterating until
+    the reviewers give their thumbs up, the "LGTM" (acronym for "Looks Good To Me").
+6.  Once the code has passed review the Pull Request can be merged into the distribution.
+
+## Pull Request Requirements
+
+In order for a Pull Request to be considered, it has to meet these requirements:
+
+1.  Live up to the current code standard:
+    - Not violate [DRY](http://programmer.97things.oreilly.com/wiki/index.php/Don%27t_Repeat_Yourself).
+    - [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) should be applied.
+2.  Be accompanied by appropriate tests.
+
+If *all* of these requirements are not met then the code should **not** be
+merged into the distribution, and need not even be reviewed.
+
+## Documentation
+
+All code contributed to the user-facing standard library (the `library/`
+directory) should come accompanied with documentation.
+Pull requests containing undocumented code will not be accepted.
+
+Code contributed to the internals (compiler, tools, JS environments, etc.)
+should come accompanied by internal documentation if the code is not
+self-explanatory, e.g., important design decisions that other maintainers
+should know about.
+
+## Creating Commits And Writing Commit Messages
+
+Follow these guidelines when creating public commits and writing commit messages.
+
+1.  If your work spans multiple local commits (for example; if you do safe point
+    commits while working in a feature branch or work in a branch for long time
+    doing merges/rebases etc.) then please do not commit it all but rewrite the
+    history by squashing the commits into **one commit per useful unit of
+    change**, each accompanied by a detailed commit message.
+    For more info, see the article:
+    [Git Workflow](http://sandofsky.com/blog/git-workflow.html).
+    Additionally, every commit should be able to be used in isolation--that is,
+    each commit must build and pass all tests.
+2.  The first line should be a descriptive sentence about what the commit is doing.
+    It should be possible to fully understand what the commit does by just
+    reading this single line.
+    It is **not ok** to only list the ticket number, type "minor fix" or similar.
+    If the commit has a corresponding ticket, include a reference to the ticket
+    number, with the format "Fix #xxx: Title of the ticket.", as the first line,
```

Comment:
That's what we've been doing most of the time, though. But true, it's not always the best message. I'll elaborate a bit.